src/BuildScriptGenerator/Python/VersionProviders/PythonOnDiskVersionProvider.cs

```diff
@@ -3,15 +3,25 @@
 // Licensed under the MIT license.
 // --------------------------------------------------------------------------------------------
 
+using Microsoft.Extensions.Logging;
+
 namespace Microsoft.Oryx.BuildScriptGenerator.Python
 {
     public class PythonOnDiskVersionProvider : IPythonVersionProvider
     {
         private const string DefaultOnDiskVersion = PythonConstants.PythonLtsVersion;
+        private readonly ILogger<PythonOnDiskVersionProvider> _logger;
+
+        public PythonOnDiskVersionProvider(ILogger<PythonOnDiskVersionProvider> logger)
+        {
+            _logger = logger;
+        }
 
         // To enable unit testing
         public virtual PlatformVersionInfo GetVersionInfo()
         {
+            _logger.LogDebug("Getting list of versions from {installDir}", PythonConstants.InstalledPythonVersionsDir);
+
```

Comment:
log the list of versions as well?

Reply:
Please take a look at my earlier comment.